src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Integer.cs

```diff
@@ -0,0 +1,166 @@
+                    AdvanceBuffer(1 + additionalBytes);
+                    return true;
+
+                default:
+                    value = 0;
+                    return false;
+            }
+        }
+
+        // Implements major type 0,1 decoding per https://tools.ietf.org/html/rfc7049#section-2.1
+        public long ReadInt64()
+        {
+            long value;
+            int additionalBytes;
+
+            ReadInitialByte(out CborDataItem header);
+
+            switch (header.MajorType)
+            {
+                case CborMajorType.UnsignedInteger:
+                    value = checked((long)ReadUnsignedInteger(in header, out additionalBytes));
+                    AdvanceBuffer(1 + additionalBytes);
+                    return value;
+
+                case CborMajorType.NegativeInteger:
+                    value = checked(-1 - (long)ReadUnsignedInteger(in header, out additionalBytes));
+                    AdvanceBuffer(1 + additionalBytes);
+                    return value;
+
+                default:
+                    throw new ArgumentException("Data item type mismatch");
+            }
+        }
+
+        public bool TryReadInt64(out long value)
+        {
+            ulong result;
+            int additionalBytes;
+
+            ReadInitialByte(out CborDataItem header);
+
+            switch (header.MajorType)
+            {
+                case CborMajorType.UnsignedInteger:
+                    result = ReadUnsignedInteger(in header, out additionalBytes);
+                    if (result > long.MaxValue)
+                    {
+                        value = 0;
+                        return false;
+                    }
+                    else
+                    {
+                        value = (long)result;
+                        AdvanceBuffer(1 + additionalBytes);
+                        return true;
+                    }
+
+                case CborMajorType.NegativeInteger:
+                    result = ReadUnsignedInteger(in header, out additionalBytes);
+                    if (result > long.MaxValue)
+                    {
+                        value = 0;
+                        return false;
+                    }
+                    else
+                    {
+                        value = -1 - (long)result;
+                        AdvanceBuffer(1 + additionalBytes);
+                        return true;
+                    }
+
+                default:
+                    value = 0;
+                    return false;
+            }
+        }
+
+        // Unsigned integer decoding https://tools.ietf.org/html/rfc7049#section-2.1
+        private ulong ReadUnsignedInteger(in CborDataItem header, out int additionalBytes)
+        {
+            switch (header.AdditionalInfo)
+            {
+                case byte x when (x < 24):
+                    additionalBytes = 0;
+                    return x;
+
+                case 24:
```

Comment:
I'd make the 24-27 be named local constants, e.g. `const int Uint8 = 24;` and then switch on those, because the magic numbers are hard to reason about.

Reply:
Done